pubber/src/main/java/daq/pubber/MqttPublisher.java

```diff
@@ -175,10 +175,13 @@ private String getMessageTopic(String deviceId, String topic) {
             + "/" + topic;
   }
 
+  @SuppressWarnings("unchecked")
   private void publishCore(String deviceId, String topicSuffix, Object data,
       Runnable callback) {
     try {
-      String payload = OBJECT_MAPPER.writeValueAsString(data);
+      String replaceWith =
```

Comment:
At this point I don't think the Object is of type "InvalidMessage" anymore, it's an message map not POJO (so no way to use instanceof to detect the type, which is why the current usage is "data instanceof Map"). I refactored the method a bit to isolate the crazy, but I don't see way to use the type subsystem to accomplish this (I wish there was, because I agree it would be better). The best I could come up with is the strongly worded object properties, so there wouldn't be any confusion when they show up.